core/templates/dev/head/pages/exploration-editor-page/improvements-tab/thread-modal/feedback-thread-modal.controller.ts

```diff
@@ -0,0 +1,76 @@
+// Copyright 2019 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Factory for creating Feedback Cards in the Improvements Tab.
+ */
+
+require(
+  'pages/exploration-editor-page/feedback-tab/services/' +
+  'thread-status-display.service.ts');
+require(
+  'pages/exploration-editor-page/feedback-tab/services/thread-data.service.ts');
+
+var oppia = require('AppInit.ts').module;
+
+oppia.controller('FeedbackThreadModalController', [
```

Comment:
Could you please use component directives instead? Standalone controllers are discouraged. You can refer to the other component directives for reference.

Also, when you do that, change the name of the file to *.directive.ts and the corresponding HTML file to *.directive.ts.

Reply:
Done.